### ISSUE TITLE
CUDA`s `dropout` doesn't support channels last memory format

### ISSUE THREAD

**Issue by VitalyFedyunin** [pytorch/pytorch#34085]
```python
x = torch.randn(200, 512, 28, 28, device='cuda', dtype=torch.float16).contiguous(memory_format=torch.channels_last)
y = torch.dropout(x, 0.5, train=True)                                                                                                                                                                                             
y.is_contiguous(memory_format=torch.channels_last) # False :(
```

cc @ngimel @VitalyFedyunin @jamesr66a

**Comment by VitalyFedyunin** [comment]
CC @jjsjann123

**Comment by VitalyFedyunin** [comment]
Looks like fixed by #33879
